problems to install

when I write make -j4 I get the next error:

src/CMakeFiles/dng-call.dir/build.make:75: recipe for target 'src/CMakeFiles/dng-call.dir/lib/call.cc.o' failed
make[2]: *** [src/CMakeFiles/dng-call.dir/lib/call.cc.o] Error 1
CMakeFiles/Makefile2:207: recipe for target 'src/CMakeFiles/dng-call.dir/all' failed
make[1]: *** [src/CMakeFiles/dng-call.dir/all] Error 2
Makefile:151: recipe for target 'all' failed
make: *** [all] Error 2

Whats my problem to be able to install your program? 

Thanks, Im looking forward to test it  !!   :)

Hi @reedacartwright   :)

Hope this helps now.

Thanks for the help!

cmake -DCMAKE_BUILD_TYPE=Release ..
-- DeNovoGear version is 1.1.1
-- Boost version: 1.68.0
-- Found the following Boost libraries:
--   program_options
--   filesystem
--   system
-- Found HTSLIB 1.9
-- Configuring done
-- Generating done
-- Build files have been written to: /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/build
cri@cri-To-be-filled-by-O-E-M:/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/build$ make -j4
[ 34%] Built target newmat
[ 65%] Built target dngutils
[ 66%] Building CXX object src/CMakeFiles/dng-call.dir/lib/call.cc.o
[ 68%] Building CXX object src/CMakeFiles/dng-call.dir/lib/pedigree.cc.o
[ 79%] Building CXX object src/CMakeFiles/dng-dnm.dir/dng-dnm.cc.o
[ 82%] Built target dng-phaser
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc: In function ‘int writeVCFHeader(std::ofstream&, std::__cxx11::string, std::__cxx11::string, std::__cxx11::string, std::__cxx11::string)’:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:13: error: no match for ‘operator==’ (operand types are ‘std::ofstream {aka std::basic_ofstream<char>}’ and ‘long int’)
   if(fo_vcf == NULL) {
             ^
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:13: note: candidate: operator==(int, long int) <built-in>
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:13: note:   no known conversion for argument 1 from ‘std::ofstream {aka std::basic_ofstream<char>}’ to ‘int’
In file included from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/contrib/newmat/newmatap.h:12:0,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/lookup.h:23,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/makeLookup.h:25,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/denovogear.h:29,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:41:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/contrib/newmat/newmat.h:2074:6: note: candidate: bool operator==(const GeneralMatrix&, const GeneralMatrix&)
 bool operator==(const GeneralMatrix& A, const GeneralMatrix& B);
      ^~~~~~~~
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/contrib/newmat/newmat.h:2074:6: note:   no known conversion for argument 1 from ‘std::ofstream {aka std::basic_ofstream<char>}’ to ‘const GeneralMatrix&’
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/contrib/newmat/newmat.h:2075:6: note: candidate: bool operator==(const BaseMatrix&, const BaseMatrix&)
 bool operator==(const BaseMatrix& A, const BaseMatrix& B);
      ^~~~~~~~
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/contrib/newmat/newmat.h:2075:6: note:   no known conversion for argument 1 from ‘std::ofstream {aka std::basic_ofstream<char>}’ to ‘const BaseMatrix&’
In file included from /usr/include/c++/6/bits/locale_conv.h:41:0,
                 from /usr/include/c++/6/locale:43,
                 from /usr/include/c++/6/iomanip:43,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/contrib/newmat/include.h:92,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/contrib/newmat/newmat.h:12,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/contrib/newmat/newmatap.h:12,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/lookup.h:23,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/makeLookup.h:25,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/denovogear.h:29,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:41:
/usr/include/c++/6/bits/unique_ptr.h:665:5: note: candidate: template<class _Tp, class _Dp> bool std::operator==(std::nullptr_t, const std::unique_ptr<_Tp, _Dp>&)
     operator==(nullptr_t, const unique_ptr<_Tp, _Dp>& __x) noexcept
     ^~~~~~~~
/usr/include/c++/6/bits/unique_ptr.h:665:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   mismatched types ‘const std::unique_ptr<_Tp, _Dp>’ and ‘long int’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/bits/locale_conv.h:41:0,
                 from /usr/include/c++/6/locale:43,
                 from /usr/include/c++/6/iomanip:43,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/contrib/newmat/include.h:92,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/contrib/newmat/newmat.h:12,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/contrib/newmat/newmatap.h:12,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/lookup.h:23,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/makeLookup.h:25,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/denovogear.h:29,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:41:
/usr/include/c++/6/bits/unique_ptr.h:660:5: note: candidate: template<class _Tp, class _Dp> bool std::operator==(const std::unique_ptr<_Tp, _Dp>&, std::nullptr_t)
     operator==(const unique_ptr<_Tp, _Dp>& __x, nullptr_t) noexcept
     ^~~~~~~~
/usr/include/c++/6/bits/unique_ptr.h:660:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::unique_ptr<_Tp, _Dp>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/bits/locale_conv.h:41:0,
                 from /usr/include/c++/6/locale:43,
                 from /usr/include/c++/6/iomanip:43,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/contrib/newmat/include.h:92,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/contrib/newmat/newmat.h:12,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/contrib/newmat/newmatap.h:12,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/lookup.h:23,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/makeLookup.h:25,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/denovogear.h:29,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:41:
/usr/include/c++/6/bits/unique_ptr.h:654:5: note: candidate: template<class _Tp, class _Dp, class _Up, class _Ep> bool std::operator==(const std::unique_ptr<_Tp, _Dp>&, const std::unique_ptr<_Up, _Ep>&)
     operator==(const unique_ptr<_Tp, _Dp>& __x,
     ^~~~~~~~
/usr/include/c++/6/bits/unique_ptr.h:654:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::unique_ptr<_Tp, _Dp>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/map:62:0,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:28:
/usr/include/c++/6/bits/stl_multimap.h:976:5: note: candidate: template<class _Key, class _Tp, class _Compare, class _Alloc> bool std::operator==(const std::multimap<_Key, _Tp, _Compare, _Alloc>&, const std::multimap<_Key, _Tp, _Compare, _Alloc>&)
     operator==(const multimap<_Key, _Tp, _Compare, _Alloc>& __x,
     ^~~~~~~~
/usr/include/c++/6/bits/stl_multimap.h:976:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::multimap<_Key, _Tp, _Compare, _Alloc>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/map:61:0,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:28:
/usr/include/c++/6/bits/stl_map.h:1311:5: note: candidate: template<class _Key, class _Tp, class _Compare, class _Alloc> bool std::operator==(const std::map<_Key, _Tp, _Compare, _Alloc>&, const std::map<_Key, _Tp, _Compare, _Alloc>&)
     operator==(const map<_Key, _Tp, _Compare, _Alloc>& __x,
     ^~~~~~~~
/usr/include/c++/6/bits/stl_map.h:1311:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::map<_Key, _Tp, _Compare, _Alloc>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/bits/stl_map.h:63:0,
                 from /usr/include/c++/6/map:61,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:28:
/usr/include/c++/6/tuple:1341:5: note: candidate: template<class ... _TElements, class ... _UElements> constexpr bool std::operator==(const std::tuple<_Args1 ...>&, const std::tuple<_Args2 ...>&)
     operator==(const tuple<_TElements...>& __t,
     ^~~~~~~~
/usr/include/c++/6/tuple:1341:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::tuple<_Args1 ...>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/tuple:39:0,
                 from /usr/include/c++/6/bits/stl_map.h:63,
                 from /usr/include/c++/6/map:61,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:28:
/usr/include/c++/6/array:241:5: note: candidate: template<class _Tp, long unsigned int _Nm> bool std::operator==(const std::array<_Tp, _Nm>&, const std::array<_Tp, _Nm>&)
     operator==(const array<_Tp, _Nm>& __one, const array<_Tp, _Nm>& __two)
     ^~~~~~~~
/usr/include/c++/6/array:241:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::array<_Tp, _Nm>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/map:60:0,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:28:
/usr/include/c++/6/bits/stl_tree.h:1282:5: note: candidate: template<class _Key, class _Val, class _KeyOfValue, class _Compare, class _Alloc> bool std::operator==(const std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>&, const std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>&)
     operator==(const _Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>& __x,
     ^~~~~~~~
/usr/include/c++/6/bits/stl_tree.h:1282:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/map:60:0,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:28:
/usr/include/c++/6/bits/stl_tree.h:328:5: note: candidate: template<class _Val> bool std::operator==(const std::_Rb_tree_iterator<_Tp>&, const std::_Rb_tree_const_iterator<_Val>&)
     operator==(const _Rb_tree_iterator<_Val>& __x,
     ^~~~~~~~
/usr/include/c++/6/bits/stl_tree.h:328:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::_Rb_tree_iterator<_Tp>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/bits/locale_facets.h:48:0,
                 from /usr/include/c++/6/bits/basic_ios.h:37,
                 from /usr/include/c++/6/ios:44,
                 from /usr/include/c++/6/ostream:38,
                 from /usr/include/c++/6/iostream:39,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:25:
/usr/include/c++/6/bits/streambuf_iterator.h:204:5: note: candidate: template<class _CharT, class _Traits> bool std::operator==(const std::istreambuf_iterator<_CharT, _Traits>&, const std::istreambuf_iterator<_CharT, _Traits>&)
     operator==(const istreambuf_iterator<_CharT, _Traits>& __a,
     ^~~~~~~~
/usr/include/c++/6/bits/streambuf_iterator.h:204:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::istreambuf_iterator<_CharT, _Traits>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/bits/ios_base.h:46:0,
                 from /usr/include/c++/6/ios:42,
                 from /usr/include/c++/6/ostream:38,
                 from /usr/include/c++/6/iostream:39,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:25:
/usr/include/c++/6/system_error:303:3: note: candidate: bool std::operator==(const std::error_condition&, const std::error_condition&)
   operator==(const error_condition& __lhs,
   ^~~~~~~~
/usr/include/c++/6/system_error:303:3: note:   no known conversion for argument 1 from ‘std::ofstream {aka std::basic_ofstream<char>}’ to ‘const std::error_condition&’
/usr/include/c++/6/system_error:296:3: note: candidate: bool std::operator==(const std::error_condition&, const std::error_code&)
   operator==(const error_condition& __lhs, const error_code& __rhs) noexcept
   ^~~~~~~~
/usr/include/c++/6/system_error:296:3: note:   no known conversion for argument 1 from ‘std::ofstream {aka std::basic_ofstream<char>}’ to ‘const std::error_condition&’
/usr/include/c++/6/system_error:289:3: note: candidate: bool std::operator==(const std::error_code&, const std::error_condition&)
   operator==(const error_code& __lhs, const error_condition& __rhs) noexcept
   ^~~~~~~~
/usr/include/c++/6/system_error:289:3: note:   no known conversion for argument 1 from ‘std::ofstream {aka std::basic_ofstream<char>}’ to ‘const std::error_code&’
/usr/include/c++/6/system_error:284:3: note: candidate: bool std::operator==(const std::error_code&, const std::error_code&)
   operator==(const error_code& __lhs, const error_code& __rhs) noexcept
   ^~~~~~~~
/usr/include/c++/6/system_error:284:3: note:   no known conversion for argument 1 from ‘std::ofstream {aka std::basic_ofstream<char>}’ to ‘const std::error_code&’
In file included from /usr/include/c++/6/string:52:0,
                 from /usr/include/c++/6/bits/locale_classes.h:40,
                 from /usr/include/c++/6/bits/ios_base.h:41,
                 from /usr/include/c++/6/ios:42,
                 from /usr/include/c++/6/ostream:38,
                 from /usr/include/c++/6/iostream:39,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:25:
/usr/include/c++/6/bits/basic_string.h:5111:5: note: candidate: template<class _CharT, class _Traits, class _Alloc> bool std::operator==(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, const _CharT*)
     operator==(const basic_string<_CharT, _Traits, _Alloc>& __lhs,
     ^~~~~~~~
/usr/include/c++/6/bits/basic_string.h:5111:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/string:52:0,
                 from /usr/include/c++/6/bits/locale_classes.h:40,
                 from /usr/include/c++/6/bits/ios_base.h:41,
                 from /usr/include/c++/6/ios:42,
                 from /usr/include/c++/6/ostream:38,
                 from /usr/include/c++/6/iostream:39,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:25:
/usr/include/c++/6/bits/basic_string.h:5099:5: note: candidate: template<class _CharT, class _Traits, class _Alloc> bool std::operator==(const _CharT*, const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&)
     operator==(const _CharT* __lhs,
     ^~~~~~~~
/usr/include/c++/6/bits/basic_string.h:5099:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   mismatched types ‘const _CharT*’ and ‘std::basic_ofstream<char>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/string:52:0,
                 from /usr/include/c++/6/bits/locale_classes.h:40,
                 from /usr/include/c++/6/bits/ios_base.h:41,
                 from /usr/include/c++/6/ios:42,
                 from /usr/include/c++/6/ostream:38,
                 from /usr/include/c++/6/iostream:39,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:25:
/usr/include/c++/6/bits/basic_string.h:5085:5: note: candidate: template<class _CharT> typename __gnu_cxx::__enable_if<std::__is_char<_Tp>::__value, bool>::__type std::operator==(const std::__cxx11::basic_string<_CharT>&, const std::__cxx11::basic_string<_CharT>&)
     operator==(const basic_string<_CharT>& __lhs,
     ^~~~~~~~
/usr/include/c++/6/bits/basic_string.h:5085:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::__cxx11::basic_string<_CharT>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/string:52:0,
                 from /usr/include/c++/6/bits/locale_classes.h:40,
                 from /usr/include/c++/6/bits/ios_base.h:41,
                 from /usr/include/c++/6/ios:42,
                 from /usr/include/c++/6/ostream:38,
                 from /usr/include/c++/6/iostream:39,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:25:
/usr/include/c++/6/bits/basic_string.h:5077:5: note: candidate: template<class _CharT, class _Traits, class _Alloc> bool std::operator==(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&)
     operator==(const basic_string<_CharT, _Traits, _Alloc>& __lhs,
     ^~~~~~~~
/usr/include/c++/6/bits/basic_string.h:5077:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/iosfwd:40:0,
                 from /usr/include/c++/6/ios:38,
                 from /usr/include/c++/6/ostream:38,
                 from /usr/include/c++/6/iostream:39,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:25:
/usr/include/c++/6/bits/postypes.h:216:5: note: candidate: template<class _StateT> bool std::operator==(const std::fpos<_StateT>&, const std::fpos<_StateT>&)
     operator==(const fpos<_StateT>& __lhs, const fpos<_StateT>& __rhs)
     ^~~~~~~~
/usr/include/c++/6/bits/postypes.h:216:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::fpos<_StateT>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/vector:64:0,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:24:
/usr/include/c++/6/bits/stl_vector.h:1514:5: note: candidate: template<class _Tp, class _Alloc> bool std::operator==(const std::vector<_Tp, _Alloc>&, const std::vector<_Tp, _Alloc>&)
     operator==(const vector<_Tp, _Alloc>& __x, const vector<_Tp, _Alloc>& __y)
     ^~~~~~~~
/usr/include/c++/6/bits/stl_vector.h:1514:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::vector<_Tp, _Alloc>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/vector:61:0,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:24:
/usr/include/c++/6/bits/allocator.h:152:5: note: candidate: template<class _Tp> bool std::operator==(const std::allocator<_Tp1>&, const std::allocator<_Tp1>&)
     operator==(const allocator<_Tp>&, const allocator<_Tp>&)
     ^~~~~~~~
/usr/include/c++/6/bits/allocator.h:152:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::allocator<_Tp1>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/vector:61:0,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:24:
/usr/include/c++/6/bits/allocator.h:146:5: note: candidate: template<class _T1, class _T2> bool std::operator==(const std::allocator<_Tp1>&, const std::allocator<_T2>&)
     operator==(const allocator<_T1>&, const allocator<_T2>&)
     ^~~~~~~~
/usr/include/c++/6/bits/allocator.h:146:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::allocator<_Tp1>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/bits/stl_algobase.h:67:0,
                 from /usr/include/c++/6/vector:60,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:24:
/usr/include/c++/6/bits/stl_iterator.h:1122:5: note: candidate: template<class _Iterator> bool std::operator==(const std::move_iterator<_IteratorL>&, const std::move_iterator<_IteratorL>&)
     operator==(const move_iterator<_Iterator>& __x,
     ^~~~~~~~
/usr/include/c++/6/bits/stl_iterator.h:1122:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::move_iterator<_IteratorL>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/bits/stl_algobase.h:67:0,
                 from /usr/include/c++/6/vector:60,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:24:
/usr/include/c++/6/bits/stl_iterator.h:1116:5: note: candidate: template<class _IteratorL, class _IteratorR> bool std::operator==(const std::move_iterator<_IteratorL>&, const std::move_iterator<_IteratorR>&)
     operator==(const move_iterator<_IteratorL>& __x,
     ^~~~~~~~
/usr/include/c++/6/bits/stl_iterator.h:1116:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::move_iterator<_IteratorL>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/bits/stl_algobase.h:67:0,
                 from /usr/include/c++/6/vector:60,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:24:
/usr/include/c++/6/bits/stl_iterator.h:352:5: note: candidate: template<class _IteratorL, class _IteratorR> bool std::operator==(const std::reverse_iterator<_Iterator>&, const std::reverse_iterator<_IteratorR>&)
     operator==(const reverse_iterator<_IteratorL>& __x,
     ^~~~~~~~
/usr/include/c++/6/bits/stl_iterator.h:352:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::reverse_iterator<_Iterator>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/bits/stl_algobase.h:67:0,
                 from /usr/include/c++/6/vector:60,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:24:
/usr/include/c++/6/bits/stl_iterator.h:295:5: note: candidate: template<class _Iterator> bool std::operator==(const std::reverse_iterator<_Iterator>&, const std::reverse_iterator<_Iterator>&)
     operator==(const reverse_iterator<_Iterator>& __x,
     ^~~~~~~~
/usr/include/c++/6/bits/stl_iterator.h:295:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::reverse_iterator<_Iterator>’
   if(fo_vcf == NULL) {
                ^
In file included from /usr/include/c++/6/bits/stl_algobase.h:64:0,
                 from /usr/include/c++/6/vector:60,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:24:
/usr/include/c++/6/bits/stl_pair.h:440:5: note: candidate: template<class _T1, class _T2> constexpr bool std::operator==(const std::pair<_T1, _T2>&, const std::pair<_T1, _T2>&)
     operator==(const pair<_T1, _T2>& __x, const pair<_T1, _T2>& __y)
     ^~~~~~~~
/usr/include/c++/6/bits/stl_pair.h:440:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/unistd.h:229:0,
                 from /usr/include/zconf.h:452,
                 from /usr/include/zlib.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/utils/bcf.h:34,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/dng-dnm.cc:164:16: note:   ‘std::ofstream {aka std::basic_ofstream<char>}’ is not derived from ‘const std::pair<_T1, _T2>’
   if(fo_vcf == NULL) {
                ^
src/CMakeFiles/dng-dnm.dir/build.make:62: recipe for target 'src/CMakeFiles/dng-dnm.dir/dng-dnm.cc.o' failed
make[2]: *** [src/CMakeFiles/dng-dnm.dir/dng-dnm.cc.o] Error 1
CMakeFiles/Makefile2:170: recipe for target 'src/CMakeFiles/dng-dnm.dir/all' failed
make[1]: *** [src/CMakeFiles/dng-dnm.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
In file included from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/include/dng/pileup.h:38:0,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/lib/call.cc:39:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/include/dng/pool.h: In instantiation of ‘dng::IntrusivePool<Node>::~IntrusivePool() [with Node = dng::pileup::Node]’:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/include/dng/pileup.h:152:12:   required from here
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/include/dng/pool.h:84:3: error: ‘const bool boost::intrusive::list_impl<boost::intrusive::bhtraits<dng::pileup::Node, boost::intrusive::list_node_traits<void*>, (boost::intrusive::link_mode_type)0u, boost::intrusive::dft_tag, 1u>, long unsigned int, true, void>::safemode_or_autounlink’ is private within this context
   if(list_type::safemode_or_autounlink)
   ^~
In file included from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/include/dng/pool.h:24:0,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/include/dng/pileup.h:38,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/lib/call.cc:39:
/usr/include/boost/intrusive/list.hpp:123:22: note: declared private here
    static const bool safemode_or_autounlink = is_safe_autounlink<value_traits::link_mode>::value;
                      ^~~~~~~~~~~~~~~~~~~~~~
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/lib/pedigree.cc: In member function ‘bool dng::Pedigree::Construct(const dng::io::Pedigree&, const dng::ReadGroups&)’:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/lib/pedigree.cc:337:64: error: call of overloaded ‘distance(std::vector<boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int> >::iterator, __gnu_cxx::__normal_iterator<boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int>*, std::vector<boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int> > >&)’ is ambiguous
     size_t num_parent_edges = distance(family_edges.begin(),pos);
                                                                ^
In file included from /usr/include/c++/6/bits/stl_algobase.h:66:0,
                 from /usr/include/c++/6/bits/char_traits.h:39,
                 from /usr/include/c++/6/string:40,
                 from /usr/include/c++/6/stdexcept:39,
                 from /usr/include/c++/6/array:39,
                 from /usr/include/c++/6/tuple:39,
                 from /usr/include/c++/6/functional:55,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/include/dng/pedigree.h:24,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/lib/pedigree.cc:20:
/usr/include/c++/6/bits/stl_iterator_base_funcs.h:135:5: note: candidate: typename std::iterator_traits<_Iterator>::difference_type std::distance(_InputIterator, _InputIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int>*, std::vector<boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int> > >; typename std::iterator_traits<_Iterator>::difference_type = long int]
     distance(_InputIterator __first, _InputIterator __last)
     ^~~~~~~~
In file included from /usr/include/boost/range/distance.hpp:18:0,
                 from /usr/include/boost/range/functions.hpp:21,
                 from /usr/include/boost/range/iterator_range_core.hpp:38,
                 from /usr/include/boost/range/iterator_range.hpp:13,
                 from /usr/include/boost/spirit/home/support/container.hpp:25,
                 from /usr/include/boost/spirit/home/support/string_traits.hpp:16,
                 from /usr/include/boost/spirit/home/support/nonterminal/expand_arg.hpp:20,
                 from /usr/include/boost/spirit/home/support/context.hpp:18,
                 from /usr/include/boost/spirit/home/qi/domain.hpp:18,
                 from /usr/include/boost/spirit/home/qi/detail/enable_lit.hpp:13,
                 from /usr/include/boost/spirit/home/qi/numeric/real.hpp:16,
                 from /usr/include/boost/spirit/include/qi_real.hpp:16,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/include/dng/utilities.h:27,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/include/dng/io/ped.h:38,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/include/dng/pedigree.h:29,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/lib/pedigree.cc:20:
/usr/include/boost/iterator/distance.hpp:49:9: note: candidate: typename boost::iterators::iterator_difference<Iterator>::type boost::iterators::distance_adl_barrier::distance(SinglePassIterator, SinglePassIterator) [with SinglePassIterator = __gnu_cxx::__normal_iterator<boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int>*, std::vector<boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int> > >; typename boost::iterators::iterator_difference<Iterator>::type = long int]
         distance(SinglePassIterator first, SinglePassIterator last)
         ^~~~~~~~
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/lib/pedigree.cc:391:58: error: call of overloaded ‘distance(std::vector<long unsigned int>::iterator, __gnu_cxx::__normal_iterator<long unsigned int*, std::vector<long unsigned int> >&)’ is ambiguous
      size_t p = distance(family_members.begin(),pivot_pos);
                                                          ^
In file included from /usr/include/c++/6/bits/stl_algobase.h:66:0,
                 from /usr/include/c++/6/bits/char_traits.h:39,
                 from /usr/include/c++/6/string:40,
                 from /usr/include/c++/6/stdexcept:39,
                 from /usr/include/c++/6/array:39,
                 from /usr/include/c++/6/tuple:39,
                 from /usr/include/c++/6/functional:55,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/include/dng/pedigree.h:24,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/lib/pedigree.cc:20:
/usr/include/c++/6/bits/stl_iterator_base_funcs.h:135:5: note: candidate: typename std::iterator_traits<_Iterator>::difference_type std::distance(_InputIterator, _InputIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<long unsigned int*, std::vector<long unsigned int> >; typename std::iterator_traits<_Iterator>::difference_type = long int]
     distance(_InputIterator __first, _InputIterator __last)
     ^~~~~~~~
In file included from /usr/include/boost/range/distance.hpp:18:0,
                 from /usr/include/boost/range/functions.hpp:21,
                 from /usr/include/boost/range/iterator_range_core.hpp:38,
                 from /usr/include/boost/range/iterator_range.hpp:13,
                 from /usr/include/boost/spirit/home/support/container.hpp:25,
                 from /usr/include/boost/spirit/home/support/string_traits.hpp:16,
                 from /usr/include/boost/spirit/home/support/nonterminal/expand_arg.hpp:20,
                 from /usr/include/boost/spirit/home/support/context.hpp:18,
                 from /usr/include/boost/spirit/home/qi/domain.hpp:18,
                 from /usr/include/boost/spirit/home/qi/detail/enable_lit.hpp:13,
                 from /usr/include/boost/spirit/home/qi/numeric/real.hpp:16,
                 from /usr/include/boost/spirit/include/qi_real.hpp:16,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/include/dng/utilities.h:27,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/include/dng/io/ped.h:38,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/include/dng/pedigree.h:29,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/lib/pedigree.cc:20:
/usr/include/boost/iterator/distance.hpp:49:9: note: candidate: typename boost::iterators::iterator_difference<Iterator>::type boost::iterators::distance_adl_barrier::distance(SinglePassIterator, SinglePassIterator) [with SinglePassIterator = __gnu_cxx::__normal_iterator<long unsigned int*, std::vector<long unsigned int> >; typename boost::iterators::iterator_difference<Iterator>::type = long int]
         distance(SinglePassIterator first, SinglePassIterator last)
         ^~~~~~~~
In file included from /media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/lib/call.cc:44:0:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/include/dng/hts/bcf.h: In member function ‘void hts::bcf::File::WriteRecord()’:
/media/cri/GEN/TRIO_TOOLS/denovogear-1.1.1/src/include/dng/hts/bcf.h:232:33: warning: ignoring return value of ‘int bcf_write(htsFile*, bcf_hdr_t*, bcf1_t*)’, declared with attribute warn_unused_result [-Wunused-result]
   bcf_write1(handle(), hdr, rec);
                                 ^
src/CMakeFiles/dng-call.dir/build.make:114: recipe for target 'src/CMakeFiles/dng-call.dir/lib/pedigree.cc.o' failed
make[2]: *** [src/CMakeFiles/dng-call.dir/lib/pedigree.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
src/CMakeFiles/dng-call.dir/build.make:75: recipe for target 'src/CMakeFiles/dng-call.dir/lib/call.cc.o' failed
make[2]: *** [src/CMakeFiles/dng-call.dir/lib/call.cc.o] Error 1
CMakeFiles/Makefile2:207: recipe for target 'src/CMakeFiles/dng-call.dir/all' failed
make[1]: *** [src/CMakeFiles/dng-call.dir/all] Error 2
Makefile:151: recipe for target 'all' failed
make: *** [all] Error 2

I did but then I get another error    ^ ^"

cmake -DCMAKE_BUILD_TYPE=Release ..
-- Compiling with build type 'Release'.
-- DeNovoGear version: 1.1.1-unknown.
-- Boost version: 1.68.0
-- Found the following Boost libraries:
--   program_options
--   filesystem
--   system
--   unit_test_framework
-- Found HTSLIB 1.9
-- Configuring done
-- Generating done
-- Build files have been written to: /media/cri/GEN/TRIO_TOOLS/denovogear-develop/build
cri@cri-To-be-filled-by-O-E-M:/media/cri/GEN/TRIO_TOOLS/denovogear-develop/build$ make -j4
[  6%] Building CXX object src/lib/CMakeFiles/libdng.dir/task/loglike.cc.o
[  6%] Building CXX object src/lib/CMakeFiles/libdng.dir/relationship_graph.cc.o
/media/cri/GEN/TRIO_TOOLS/denovogear-develop/src/lib/relationship_graph.cc: In member function ‘void dng::RelationshipGraph::CreatePeelingOps(const Graph&, const std::vector<long unsigned int>&, dng::RelationshipGraph::family_labels_t&, const pivots_t&)’:
/media/cri/GEN/TRIO_TOOLS/denovogear-develop/src/lib/relationship_graph.cc:1065:69: error: call of overloaded ‘distance(std::vector<boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int> >::iterator, __gnu_cxx::__normal_iterator<boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int>*, std::vector<boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int> > >&)’ is ambiguous
         size_t num_spousal_edges = distance(family_edges.begin(), it);
                                                                     ^
In file included from /usr/include/c++/6/bits/stl_algobase.h:66:0,
                 from /usr/include/c++/6/bits/char_traits.h:39,
                 from /usr/include/c++/6/string:40,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-develop/src/lib/../include/dng/relationship_graph.h:26,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-develop/src/lib/relationship_graph.cc:20:
/usr/include/c++/6/bits/stl_iterator_base_funcs.h:135:5: note: candidate: typename std::iterator_traits<_Iterator>::difference_type std::distance(_InputIterator, _InputIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int>*, std::vector<boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int> > >; typename std::iterator_traits<_Iterator>::difference_type = long int]
     distance(_InputIterator __first, _InputIterator __last)
     ^~~~~~~~
In file included from /usr/include/boost/range/distance.hpp:18:0,
                 from /usr/include/boost/range/functions.hpp:21,
                 from /usr/include/boost/range/iterator_range_core.hpp:38,
                 from /usr/include/boost/range/iterator_range.hpp:13,
                 from /usr/include/boost/range/as_literal.hpp:22,
                 from /usr/include/boost/algorithm/string/predicate.hpp:19,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-develop/src/lib/../include/dng/library.h:27,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-develop/src/lib/../include/dng/relationship_graph.h:30,
                 from /media/cri/GEN/TRIO_TOOLS/denovogear-develop/src/lib/relationship_graph.cc:20:
/usr/include/boost/iterator/distance.hpp:49:9: note: candidate: typename boost::iterators::iterator_difference<Iterator>::type boost::iterators::distance_adl_barrier::distance(SinglePassIterator, SinglePassIterator) [with SinglePassIterator = __gnu_cxx::__normal_iterator<boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int>*, std::vector<boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int> > >; typename boost::iterators::iterator_difference<Iterator>::type = long int]
         distance(SinglePassIterator first, SinglePassIterator last)
         ^~~~~~~~
src/lib/CMakeFiles/libdng.dir/build.make:179: recipe for target 'src/lib/CMakeFiles/libdng.dir/relationship_graph.cc.o' failed
make[2]: *** [src/lib/CMakeFiles/libdng.dir/relationship_graph.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:331: recipe for target 'src/lib/CMakeFiles/libdng.dir/all' failed
make[1]: *** [src/lib/CMakeFiles/libdng.dir/all] Error 2
Makefile:162: recipe for target 'all' failed
make: *** [all] Error 2

--- a/src/lib/relationship_graph.cc
+++ b/src/lib/relationship_graph.cc
@@ -1062,7 +1062,7 @@ void dng::RelationshipGraph::CreatePeelingOps(
         auto it = boost::find_if(family_edges, [&](edge_t x) -> bool {
             return (edge_types(x) != EdgeType::Spousal);
         });
-        size_t num_spousal_edges = distance(family_edges.begin(), it);
+        size_t num_spousal_edges = std::distance(family_edges.begin(), it);
 
         // Check to see what type of graph we have
         if (num_spousal_edges == 0) {

